File: forge/src/main/java/com/selfdot/cobblemontrainers/forge/CobblemonTrainersForge.java
package com.selfdot.cobblemontrainers.forge;

import com.selfdot.cobblemontrainers.CobblemonTrainers;
import net.minecraftforge.api.distmarker.Dist;
import net.minecraftforge.common.MinecraftForge;
import net.minecraftforge.eventbus.api.IEventBus;
import net.minecraftforge.fml.DistExecutor;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.event.lifecycle.FMLCommonSetupEvent;
import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;

@Mod(CobblemonTrainers.MODID)
public class CobblemonTrainersForge {

    public CobblemonTrainersForge() {
        IEventBus MOD_BUS = FMLJavaModLoadingContext.get().getModEventBus();
        MOD_BUS.addListener(this::commonSetup);
        MinecraftForge.EVENT_BUS.register(this);
    }

    private void commonSetup(final FMLCommonSetupEvent event) {
        CobblemonTrainers.INSTANCE.initialize();
        DistExecutor.safeRunWhenOn(Dist.DEDICATED_SERVER, SetPermissionValidatorRunnable::new);
        System.out.println("CobblemonTrainers Forge initialized");
    }

}

File: forge/src/main/kotlin/com/selfdot/cobblemontrainers/forge/CobblemonTrainersForge.kt
package com.selfdot.cobblemontrainers.forge

import dev.architectury.platform.forge.EventBuses
import com.selfdot.cobblemontrainers.CobblemonTrainers
import java.util.*
import net.minecraftforge.fml.common.Mod
import net.minecraftforge.fml.event.lifecycle.FMLCommonSetupEvent
import net.minecraftforge.fml.event.lifecycle.FMLDedicatedServerSetupEvent

@Mod(CobblemonTrainers.MODID)
class CobblemonTrainersForge {

    init {
        with(thedarkcolour.kotlinforforge.forge.MOD_BUS) {
            EventBuses.registerModEventBus(CobblemonTrainers.MODID, this)
            addListener(this@CobblemonTrainersForge::initialize)
            addListener(this@CobblemonTrainersForge::serverInit)
        }
    }

    private fun serverInit(event: FMLDedicatedServerSetupEvent) { }

    private fun initialize(event: FMLCommonSetupEvent) {
        CobblemonTrainers.initialize()
        CobblemonTrainers.permissionValidator = ForgePermissionValidator()
        println("CobblemonTrainers Forge initialized")
    }

}
